src/app/components/index/index.component.ts

```diff
@@ -145,11 +150,52 @@ export class IndexComponent implements OnInit {
   }
 
   handleCompetitionDeleteClick(competitionId: number) {
-    this.competitionService.deleteCompetition(competitionId).subscribe(() => {
-      this.competitionService.getCompetitions().subscribe((competitions) => {
-        this.competitions = competitions;
-        this.selectCompetitions();
-      });
+    const dialogRef = this.dialog.open(ConfirmDialogComponent, {
+      data: {
+        title: 'Delete competition',
+        content: 'Are you sure you want to delete this competition?',
+      },
+    });
+    dialogRef.afterClosed().subscribe((result) => {
+      if (result) {
+        this.competitionService
+          .deleteCompetition(competitionId)
+          .subscribe(() => {
+            this.competitionService
+              .getCompetitions()
+              .subscribe((competitions) => {
+                this.competitions = competitions;
+                this.selectCompetitions();
+              });
+          });
+      }
+    });
+  }
+
+  handleEventViewClick(eventId: number) {
+    this.router.navigate(['/events/', eventId]);
+  }
+
+  handleEventEditClick(eventId: number) {
+    this.router.navigate(['/events/edit/', eventId]);
+  }
+
+  handleEventDeleteClick(eventId: number) {
+    const dialogRef = this.dialog.open(ConfirmDialogComponent, {
+      data: {
+        title: 'Delete event',
+        content: 'Are you sure you want to delete this event?',
+      },
+    });
+    dialogRef.afterClosed().subscribe((result) => {
+      if (result) {
+        this.eventService.deleteEvent(eventId).subscribe(() => {
+          this.eventService.getEvents().subscribe((events) => {
```

Comment:
if you call  this.eventService.getEvents() just to update your component's events array you can just splice the actual array after the delete request is done instead of making another http request